### PR TITLE
Add biggest bust / surprise methods to class Season

### PR DESCRIPTION
**Pull request by AlexisDumortier**
- added the methods biggest_bust & biggest_surprise to the Season class (worked on these methods with Pierce)
- refactored the winningest_coach and worst_coach to remove unused calculations

### CODE REVIEW

File: lib/season.rb [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 
+  attr_reader :game_collection
+
   def initialize(csv_file_path = './test/fixtures/games.csv')
     @game_collection  = GameCollection.new(csv_file_path)
   end
 
   def games_per_season
-    @game_collection.group_by{|game| game.season}
+    @game_collection.games.group_by { |game| game.season }
```
Comment (AlexisDumortier) [inline lib/season.rb]: fixed this -> `group_by` could not be called directly on a `game_collection` object.

File: lib/season.rb [hunk 2]
```diff
@@ -97,4 +100,81 @@ def fewest_tackles(season_id)
     Team.team_id_to_team_name(fewest_tackles_team_id)
   end
 
+  def games_per_season_type(games, season_type)
```
Comment (AlexisDumortier) [inline lib/season.rb]: moved these methods from the GameCollection class

File: test/season_test.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 require 'minitest/autorun'
 require 'minitest/pride'
 require_relative '../lib/season'
+require 'mocha/minitest'
```
Comment (AlexisDumortier) [inline test/season_test.rb]: added this line to allow stubbing some of the methods

File: test/season_test.rb [hunk 4]
```diff
@@ -54,7 +55,110 @@ def test_fewest_tackles_by_team_by_season
...
+      "28" => team_28,
+      "29" => team_29,
+      "30" => team_30
+    }
+    expected = {
+      "3" => 0.1, "5" => 0.75, "6" => 1.0, "8" => 0.0, "9" => 1.0, "13" => 0.0,
+      "14" => 0.5, "15" => 1.0, "16" => 0.714, "19" => 0.0, "26" => 0.4, "28" => 0.6, "29" => 1.0, "30" => 0.2
+    }
+
+    @season.expects(:win_log_per_team).with(games_log_per_team).returns(
```
Comment (AlexisDumortier) [inline test/season_test.rb]: stubbing out the `win_log_per_team` method

Comment (palworth) [reply test/season_test.rb]: good implementation of stub.

File: lib/season.rb (hunk 1 above)
Comment (palworth) [reply lib/season.rb]: Syntactic sugar.

**Pull request merged**